core/src/main/java/org/apache/carbondata/core/datastore/block/TableBlockInfo.java

```diff
@@ -306,7 +306,7 @@ public boolean equals(Object obj) {
       return  false;
     }
 
-    if (!filePath.equals(other.filePath)) {
+    if (!new Path(filePath).equals(new Path(other.filePath))) {
```

Comment:
Why this is required? please add comment. 
Is there a more efficient way to do it without create new objects

Reply:
will move to another pr